flow/Hostname.actor.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * Hostname.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "flow/Hostname.h"
+#include "flow/UnitTest.h"
+#include "flow/actorcompiler.h" // has to be last include
+
+Hostname Hostname::parse(const std::string& s) {
+	if (s.empty() || !Hostname::isHostname(s)) {
+		throw connection_string_invalid();
```

Comment:
Should this still be connction_string_invalid()? Or something like hostname_invalid().

Can be addressed in a follow up PR.